### DISCUSSION TITLE
Formula installs python version it depends on, but `six` package with the same code does not

### DISCUSSION THREAD

**Discussion by Oleksii-Svintsitskyi** [Homebrew/discussions#3540] [Writing Formulae/Casks] [answered]
Why when I'm creating my own formula without any additional resources and use
```
depends_on "python@3.10" => [:build, :test] 
depends_on "python@3.8" => [:build, :test]
depends_on "python@3.9" => [:build, :test]

def pythons
  deps.map(&:to_formula).sort_by(&:version)
end
  
def install
  pythons.each do |python|
    python_exe = python.opt_libexec/"bin/python"
    system python_exe, *Language::Python.setup_install_args(prefix, python)
end
```
just like in `six` package, my Formula installs all the python versions that is depends on?

And when I'm install `six` package, it just installs the package for all the pythons (3.8, 3.9, 3.10) that are already installed, but not installs pythons itself?

**Comment by SMillerDev** [comment, marked as answer]
It'll only install those pythons if you build from source

**Reply by Oleksii-Svintsitskyi** [reply]
@SMillerDev Do you mean `brew install --build-from-source <package>`?
If yes, than it's not true in this situation, because I'm just running `brew install <package>`

**Reply by Oleksii-Svintsitskyi** [reply]
@SMillerDev or it has something to do with the `bottle` section?

**Reply by SMillerDev** [reply]
If there are no prebuilt binaries it'll build from source.

**Reply by Oleksii-Svintsitskyi** [reply]
@SMillerDev thank you!
